internal/xds/translator/translator.go

```diff
@@ -78,6 +78,46 @@ func Translate(ir *ir.Xds) (*types.ResourceVersionTable, error) {
 		tCtx.AddXdsResource(resource.RouteType, xdsRouteCfg)
 	}
 
+	for _, tlsListener := range ir.TLS {
+		// 1:1 between IR TLSListener and xDS Listener
+		xdsListener, err := buildXdsPassthroughListener(tlsListener)
+		if err != nil {
+			return nil, multierror.Append(err, errors.New("error building xds listener"))
+		}
+
+		// Allocate virtual host for this tlsListener.
```

Comment:
are you sure we need a route resource ?
looking at https://github.com/envoyproxy/envoy/pull/13975 you probably need a filter chain match to match the SNI
```
    - filter_chain_match:
         server_names:
         - domain3.example.com
```
and directly assign a `cluster` to the filter
cc @lizan

Reply:
addressed. could you PTAL. thanks